### ISSUE TITLE
Frequencies with different dertype settings differ more than expected

### ISSUE THREAD

**Issue by maaren** [psi4/psi4#901]
While trying to reproduce some gaussian calculations with psi4, I noticed that the frequencies dependent on the dertype setting to a larger extend than anticipated: 
```
molecule h2o {
units angstrom
H  0.000000  0.760300  -0.196100
O  0.000000  0.000000  0.392200
H  0.000000  -0.760300  -0.196100
}

set    basis 6-311G**
set e_convergence 9
set g_convergence gau_verytight
set scf_type pk

optimize('scf', molecule=h2o)

scf_e, scf_wfn = frequencies('scf', dertype='energy', return_wfn=True)
scf_e, scf_wfn = frequencies('scf', dertype='gradient', return_wfn=True)
scf_e, scf_wfn = frequencies('scf', dertype='hessian', return_wfn=True)
```
This generated the following frequencies:
```
$ grep -A 3 'Vib. Freq' 'hf-6-311Gstarstar-freq.out'
  No.    Vib. Freq. [cm^-1]      Vib. Temp. [K]
    1              1750.943            2519.214
    2              4142.104            5959.557
    3              4237.400            6096.666
--
  No.    Vib. Freq. [cm^-1]      Vib. Temp. [K]
    1              1750.925            2519.188
    2              4142.111            5959.567
    3              4237.427            6096.704
--
  No.    Vib. Freq. [cm^-1]      Vib. Temp. [K]
    1              1731.766            2491.623
    2              4136.243            5951.124
    3              4232.474            6089.579
```
For reference these are the gaussian g09 frequencies:
```
Frequencies -- 1750.9454 4142.1029 4237.3716
```

I am using:
```
$ rpm -qif /usr/bin/psi4 | egrep '^Version|^Source'
Version : 1.1
Source RPM : psi4-1.1-2.add49b9git.fc26.src.rpm
```

**Comment by loriab** [comment]
Acknowledged. This (differences between findif and analytic vibs) and handling of  trans/rot spaces for vibs at non-stationary geometries are the primary holdups to psi4/psi4#834.

**Comment by andysim** [comment]
It took me getting flu to find the time to look into this, and here's my initial finding.  If I optimize the geometry and dump it into an input file
```
molecule h2o {
    O            0.000000000000    -0.000000000000     0.063771204693
    H           -0.000000000000     0.748825894352    -0.506047648051
    H            0.000000000000    -0.748825894352    -0.506047648051
}

set    basis 6-311G**
set e_convergence 9
set g_convergence gau_verytight
set scf_type pk

scf_e, scf_wfn = frequencies('scf', dertype='hessian', return_wfn=True)
```
I get 
```  No.    Vib. Freq. [cm^-1]      Vib. Temp. [K]
    1              1750.945            2519.217
    2              4142.103            5959.555
    3              4237.372            6096.625
```
If I permute the first two rows of the geometry, per the bug report, I get:
```  No.    Vib. Freq. [cm^-1]      Vib. Temp. [K]
    1              1731.766            2491.623
    2              4136.243            5951.124
    3              4232.474            6089.579
```
This is clearly wrong, but it does at least mean that the hessian code is for the most part correct.   I'm trying to figure out where the atom and/or basis function ordering issue is introduced and hope to have a patch soon.  Thanks for the report.

**Comment by andysim** [comment]
This should be fixed in #903 (the comment in that PR was supposed to close this ticket, but didn't for some reason).  Please ping back if you have any more problems.
